sympy/physics/control/tests/test_lti.py

```diff
@@ -1748,3 +1748,135 @@ def test_StateSpace_functions():
     assert ss3.input_matrix == Matrix([[0, 0], [1, 0], [0, 1], [0, 0]])
     assert ss3.output_matrix == Matrix([[0, 1, 0, 0], [0, 0, 1, 0]])
     assert ss3.feedforward_matrix == Matrix([[0, 0], [0, 1]])
+
+
+def test_StateSpace_series():
+    # For SISO Systems
+    A1 = Matrix([[0, 1], [1, 0]])
```

Comment:
Hi, where are these tests from ? If you can you link any references I'll go through them.
I'd suggest it is good practice to test out `MATLAB` examples as well as then you can judge accuracy of your outputs.

Reply:
Hi @faze-geek I've found a issue with the current implementation of multiplication for state space objects.
### Matlab Code
```
a1 = [0, 1; 1, 0];
b1 = [0; 1];
c1 = [0, 1];
d1 = 0;
ss1 = ss(a1, b1, c1, d1);
a2 = [1,0;0,1];
b2 = [1;0];
c2 = [1,0];
d2 = 1;
ss2 =ss(a2, b2, c2, d2);
ser1 = series(ss2, ss1);
```
Output
```
>> ss1 * ss2

ans =
 
  A = 
       x1  x2  x3  x4
   x1   0   1   0   0
   x2   1   0   1   0
   x3   0   0   1   0
   x4   0   0   0   1
 
  B = 
       u1
   x1   0
   x2   1
   x3   1
   x4   0
 
  C = 
       x1  x2  x3  x4
   y1   0   1   0   0
 
  D = 
       u1
   y1   0

>> ss2 * ss1

ans =
 
  A = 
       x1  x2  x3  x4
   x1   1   0   0   1
   x2   0   1   0   0
   x3   0   0   0   1
   x4   0   0   1   0
 
  B = 
       u1
   x1   0
   x2   0
   x3   0
   x4   1
 
  C = 
       x1  x2  x3  x4
   y1   1   0   0   1
 
  D = 
       u1
   y1   0

```
### SymPy Multiplication
```
>>> a1 = Matrix([[0, 1], [1, 0]])
>>> b1 = Matrix([[0], [1]])
>>> c1 = Matrix([[0, 1]])
>>> d1 = Matrix([[0]])
>>> a2 = Matrix([[1, 0], [0, 1]])
>>> b2 = Matrix([[1], [0]])
>>> c2 = Matrix([[1, 0]])
>>> d2 = Matrix([[1]])
>>> ss1 = StateSpace(a1, b1, c1, d1)
>>> ss2 = StateSpace(a2, b2, c2, d2)
>>> ss1*ss2
StateSpace(
Matrix([
[1, 0, 0, 0],
[0, 1, 0, 0],
[0, 0, 0, 1],
[1, 0, 1, 0]]),

Matrix([
[1],
[0],
[0],
[1]]),

Matrix([[0, 0, 0, 1]]),

Matrix([[0]]))
>>> ss2*ss1
StateSpace(
Matrix([
[0, 1, 0, 0],
[1, 0, 0, 0],
[0, 1, 1, 0],
[0, 0, 0, 1]]),

Matrix([
[0],
[1],
[0],
[0]]),

Matrix([[0, 1, 1, 0]]),

Matrix([[0]]))
```

Reply:
I think there is some bug present in multiplication of the `StateSpace` which is causing problem.

Reply:
You would be aware that `StateSpaceModel` is not a unique right ? Meaning one input output system can have infinite `StateSpace` representations unlike `TransferFunctionModel`. Similarly the output here might be different but not necessarily incorrect.

Reply:
Consider this in python-control - 

```
import numpy as np
import control

a1 = np.array([[0, 1], [1, 0]])
b1 = np.array([[0], [1]])
c1 = np.array([[0, 1]])
d1 = np.array([[0]])

ss1 = control.ss(a1, b1, c1, d1)

a2 = np.array([[1, 0], [0, 1]])
b2 = np.array([[1], [0]])
c2 = np.array([[1, 0]])
d2 = np.array([[1]])

ss2 = control.ss(a2, b2, c2, d2)
```

Reply:
Output -
```
print(ss1*ss2)

A = [[1. 0. 0. 0.]
     [0. 1. 0. 0.]
     [0. 0. 0. 1.]
     [1. 0. 1. 0.]]

B = [[1.]
     [0.]
     [0.]
     [1.]]

C = [[0. 0. 0. 1.]]

D = [[0.]]

```
```
print(ss2*ss1)

A = [[0. 1. 0. 0.]
     [1. 0. 0. 0.]
     [0. 1. 1. 0.]
     [0. 0. 0. 1.]]

B = [[0.]
     [1.]
     [0.]
     [0.]]

C = [[0. 1. 1. 0.]]

D = [[0.]]
```

Reply:
This matches with the SymPy results. I'm not sure what kind of manipulations MATLAB does but I've observed it always applies minimal realization (https://github.com/sympy/sympy/issues/25340) on it's systems which is not followed in SymPy, so we can have different outputs in some instances.